lib/addons/cluster-autoscaler/index.ts

```diff
@@ -72,30 +76,39 @@ export class ClusterAutoScalerAddOn extends HelmAddOn {
             "autoscaling:DescribeTags",
             "autoscaling:SetDesiredCapacity",
             "autoscaling:TerminateInstanceInAutoScalingGroup",
-            "ec2:DescribeInstanceTypes",
             "ec2:DescribeLaunchTemplateVersions"
         );
-        const autoscalerPolicy = new iam.Policy(cluster.stack, "cluster-autoscaler-policy", {
-            policyName: "ClusterAutoscalerPolicy",
-            statements: [autoscalerStmt],
+
+        const autoscalerPolicyDocument = new iam.PolicyDocument({
+            statements: [autoscalerStmt]
         });
+
+        // Tag node groups
         const clusterName = new CfnJson(cluster.stack, "clusterName", {
             value: cluster.clusterName,
         });
-        
         for(let ng of nodeGroups) {
-            autoscalerPolicy.attachToRole(ng.role);
             Tags.of(ng).add(`k8s.io/cluster-autoscaler/${clusterName}`, "owned", { applyToLaunchedInstances: true });
             Tags.of(ng).add("k8s.io/cluster-autoscaler/enabled", "true", { applyToLaunchedInstances: true });
         }
 
-        const clusterAutoscalerChart = this.addHelmChart(clusterInfo, {
-            cloudProvider: 'aws',
-            autoDiscovery: {
-                clusterName: cluster.clusterName
-            },
-            awsRegion: clusterInfo.cluster.stack.region
-        });
+        // Create IRSA
+        const sa = createServiceAccount(cluster,
+            RELEASE, 
+            "kube-system",
```

Comment:
This should use the namespace provided by the customer.

Reply:
addressed